exporter/opentelemetry-exporter-otlp-proto-grpc/src/opentelemetry/exporter/otlp/proto/grpc/exporter.py

```diff
@@ -204,7 +204,9 @@ def __init__(
         endpoint: Optional[str] = None,
         insecure: Optional[bool] = None,
         credentials: Optional[ChannelCredentials] = None,
-        headers: Optional[Sequence] = None,
+        headers: Union[
+            TypingSequence[Tuple[str, str]], Dict[str, str], None
```

Comment:
Could there be encoded values passed in directly (not via env var) into the exporter?

Reply:
Technically this class **can** be initialized by passing `headers` as a `str`.
That's the only case (IMO) where someone would  pass encoded values (and those will get processed by `parse_headers`).
We could add `str` to the `Union` in the typedef if you think that it should be "officially" supported.

When `headers` are provided as `Sequence[Tuple[str, str]]` or `Dict[str, str]` there is currently an assumption that the explicit values are in the correct format.
For example, the keys are not lowercased (although gRPC requires it).
I opted to not manipulate explicit input here but am open to opinions!

Reply:
I believe accepting env vars already implies `str` is officially supported so it should be included in the typing. 

> When headers are provided as Sequence[Tuple[str, str]] or Dict[str, str] there is currently an assumption that the explicit values are in the correct format.

I think to be safe and consistent, it would be good to apply `parse_headers` in all cases.

Reply:
The type hint is an easy change that I can push up.
The hint is getting pretty big, happy to change it to an alias if that's preferred!

---

> I think to be safe and consistent, it would be good to apply parse_headers in all cases.

I'm not sure what the implementation of `parse_headers` for all cases would look like.

If it's just the normalization (ie. `strip` and `lower`) -- I can apply those to the dict/tuple args.

The rest of `parse_headers` performs pattern matching or unquoting.
I think it'd be unintuitive to do matching against `_KEY_VALUE_FORMAT` or `_VALUE_FORMAT` when passing a `Dict[str, str]` or `Sequence[Tuple[str, str]]`.


```py
# This applies *part* of the baggage serialization spec
# but we already have discrete keys/values.
# This is also immediately getting unquoted again.
headers={"X-Example": "value%20with%20spaces"}

# This is "what-you-see-is-what-you-get".
headers={"X-Example": "value with spaces"}
```

Reply:
Yeah, I don't have a strong opinion on this. I'm okay with just leaving it as is under the assumption that `Dict` and `Tuple` values are in the correct format.